core/shared/src/main/scala/org/http4s/internal/package.scala

```diff
@@ -329,4 +330,13 @@ package object internal {
       }
     }
   }
+
+  private[http4s] def javaMajorVersion[F[_]](implicit F: Sync[F]): F[Option[Int]] =
```

Comment:
This seems unused. Do we want this within `0.23`?

Reply:
it's used in 0.22.  Something like it will need to be in Cats-Effect if we fix its Random.  I left it mainly because I don't want to write it again if we ever need it, but it wasn't hard.  I guess we can prune it.